src/components/top-bar.tsx

```diff
@@ -47,47 +47,42 @@ export class TopBarComponent extends BaseComponent<IProps, IState> {
                  onClick={this.handleClickRightMagnetRemoveButton}>Remove magnet</div>
           </div>
         </div>
-        {this.renderMagnetButton(secondaryMagText, secondaryMagType, "right", this.handleClickRightMagnetButton)}
+        {this.renderMagnetButton(secondaryMagText, secondaryMagType, "right")}
       </div>
     );
   }
 
-  private renderMagnetButton = (text: string, magType: any, posClass: string, onClickFunction: () => void) => {
-    const magImage = "assets/magnet-" + magType + "-icon.png";
+  private renderMagnetButton = (text: string, magType: MagnetType | null, posClass: string) => {
     return (
-      <div className={"magnet-button " + posClass} onClick={onClickFunction}>
-      {magType === "none"
-        ? <div className={"unselected " + posClass}>{text}</div>
-        : <img src={magImage} className={"icon " + posClass}/>}
+      <div className={"magnet-button " + posClass} onClick={this.handleClickMagnetButton}>
+      {magType
+        ? <img src={"assets/magnet-" + magType + "-icon.png"} className={"icon " + posClass}/>
+        : <div className={"unselected " + posClass}>{text}</div>}
       </div>
     );
   }
 
-  private renderBarButton = (magType: any, disabledClass: string, onClickFunction: () => void) => {
+  private renderBarButton = (magType: MagnetType | null, disabledClass: string, onClickFunction: () => void) => {
```

Comment:
Ah it seems like using `null` instead of `"none"` made some of the logic more confusing...but still, I think this all looks fine

Reply:
Agreed.  Not sure I'm crazy about pulling it out since it adds cases like this.

Reply:
I think as long as we remember that a magnet can be bar, coil, or null we should be fine.